core/src/main/java/io/atomix/core/utils/config/PolymorphicConfigMapper.java

```diff
@@ -97,7 +98,7 @@ protected void checkRemainingProperties(Set<String> missingProperties, List<Stri
         .filter(propertyName -> !isPolymorphicType(clazz) || !polymorphicTypes.stream().anyMatch(type -> Objects.equals(type.getTypePath(), propertyName)))
         .map(propertyName -> toPath(path, propertyName))
         .filter(propertyName -> !properties.containsKey(propertyName))
```

Comment:
properties is also accessed in the filter call above, wouldn't it make sense making a new Properties above? eg..

```java
Properties localProperties = new Properties();
localProperties.putAll(System.getProperties());
```

Reply:
You're right, thanks